packages/shared/src/components/MainFeedLayout.tsx

```diff
@@ -295,6 +300,19 @@ export default function MainFeedLayout({
   const disableTopPadding =
     isFinder || shouldUseMobileFeedLayout || shouldUseCommentFeedLayout;
 
+  const { openModal } = useLazyModal();
+  const { shouldShowPopup } = useStreakExperiment();
+  const { completeAction } = useActions();
+
+  useEffect(() => {
+    if (!shouldShowPopup) {
+      return;
+    }
+
+    openModal({ type: LazyModal.MigrateUserStreak });
+    completeAction(ActionType.ExistingUserSeenStreaks);
```

Comment:
Just a mindful check, would u mind checking with product if they want to add any tracking about this popup?

Reply:
Cool. Let me check.

Reply:
No additional tracking is necessary [as per Sab](https://dailydotdev.slack.com/archives/C05TQL6HWTH/p1713262800048729?thread_ts=1713182078.933259&cid=C05TQL6HWTH).